server/src/main/java/keywhiz/auth/bcrypt/BcryptAuthenticator.java

```diff
@@ -48,7 +52,7 @@ public BcryptAuthenticator(UserDAO userDAO) {
     String password = credentials.getPassword();
 
     // Get hashed password column from BCrypt table by username
-    Optional<String> optionalHashedPwForUser = userDAO.getHashedPassword(username);
+    Optional<String> optionalHashedPwForUser = userDAO.getHashedPassword(dslContext, username);
```

Comment:
This example makes me think changing the API like this may be a code smell. UserDAO requires DSLContext for its lookup (`getHashedPassword`) and its not different based on the lookup. The DSLContext is really a dependency of UserDAO not BcryptAuthenticator, so it really doesn't belong here.

Formerly with JDBI, there were still transaction scopes but they were bound to a Handle object. There was some magic using the (SQL Object API)[http://jdbi.org/sql_object_overview/], which made it transparent. What happened is that instantiated DAO objects had the handle passed in as a dependency. To handle the case of a transaction wrapping multiple calls across classes, CreateSqlObject would pass the current handle to the other DAO instead of a new one. Ultimately, code outside of DAOs wouldn't know about Handle objects, and similarly maybe they shouldn't know about DSLContext objects.